python/modules/map_plate.py

```diff
@@ -7,30 +7,65 @@
 logger = logging.getLogger("map plate")
 logger.info("Executing map_plate module.")
 
-def _preparing_output_csv(input_data, output_path, mp_data, names, mp_deltimer, csv_deltimer):
-    output_data = []
-    num = 0
-    position = 0 
-    for line in input_data:
-        if num == 0: 
-            new_line = line
-            for i in range(len(line)):
-                if line[i] == "well.name":
-                    position = i
-            for j in names:
-                new_line.append(j)
-            output_data.append(new_line)
-            num = 1
+def parse_mp(input_path, delimiter):
+    mp_dict = {}
+    dim =_parse_base_params(input_path, delimiter, mp_dict)
+    _parse_params(input_path, delimiter, dim, mp_dict)
+    return mp_dict
+
+def _parse_params(input_path, delimiter, dimensions, mp_dict):
+    param_paths =_get_param_paths(input_path)
+    param_paths = sorted(param_paths, key = str)
+    logger.debug("Map_plate params input paths: %s.", param_paths)
+    for mp_file in param_paths:
+        correctness = CSV_M.verify_input_file(mp_file, delimiter, dimensions)
+        if correctness == False:
+            logger.error("Fatal error occured while trying to read mapplate.")
+            exit()
         else:
-            new_line = line
-            key = line[position]
-            mp_data_line = mp_data[key]
-            mp_data_line = mp_data_line.replace(mp_deltimer, csv_deltimer)
-            new_line.append(mp_data_line)
-            output_data.append(new_line)
-    return output_data
+            _collect_exp_settings(input_path, mp_file, delimiter, mp_dict)
+
+def _parse_base_params(input_path, delimiter, mp_dict):
+    active_path = FM.path_join(input_path, "args_active.csv")
```

Comment:
@stork119  "Base" params are the params which need to be present in every map_plate: args_active, args_ind, args_name, "not base" params are other information that are facultative and not pre-defined.

I guess I can extract args_active part and verification of args_name and id.